File: python/10773.py
# import sys
# input = sys.stdin.readline
# 이것을 사용하면 시간 단축 가능

k = int(input())

a = []

for _ in range(k):
    b = int(input())
    if (b == 0):
        a.pop()
    else:
        a.append(b)

print(sum(a))

File: c++/10773.cpp
#include <iostream>
#include <stack>

using namespace std;

int main(){
  int k;
  stack<int> st;
  cin >> k;

  int sum = 0;
  for(int i=0; i<k; i++){
    int a;
    cin >> a;
    if(a == 0){
      sum -= st.top();
      st.pop();
    }else{
      st.push(a);
      sum += st.top();
    }
  }
  
  cout << sum;
  return 0;
}